### ISSUE TITLE
Chef provisioner treating variables as strings when starting chef run

### ISSUE THREAD

**Issue by ivanst** [hashicorp/terraform#2149]
Hi. I've written a module that has the chef provisioner in it. I pass variables into the provisioner block of code but they are passed directly to the server commands as strings instead of the contents of the variable. For example:

```js
resource "aws_instance" "generic" {
  ami = "${lookup("${var.get_ami_id}", concat("${var.operating_system}", "-", "${var.region}"))}"
  instance_type = "${var.instance_type}"
  key_name = "${var.key_name}"
  availability_zone = "${var.az}"

  associate_public_ip_address = "${var.public}"
  subnet_id = "${var.subnet_id}"
  security_groups = ["${split(",", "${var.sgroups}")}"]

  tags { Name = "${var.name}" }
  root_block_device { delete_on_termination = true }

  count = "${var.count}"

  connection {
    agent = true
    user = "${var.ssh_user}"
  }

  module "chef_provisioner" {

  }
  provisioner "chef" {
    environment = "${var.env}"
    run_list = ["${split(",", "${var.runlists}")}"]
    node_name = "${var.name}"
    server_url = "https://api.opscode.com/organizations/foo"
    validation_client_name = "foo-validator"
    validation_key_path = "../../shared/bootstrap/foo-validator.pem"
    version = "${var.chef_version}"
  }
}
```

I get the error below when it brings the instance up. Note the variable dumped directly into the command as a string rather than the value assigned:

```
* Command "sudo bash ./install.sh -v ${var.chef_version}" exited with non-zero exit status: 1
```

After hardcoding chef_version, this error followed:

```
* Command "sudo chef-client -j \"/etc/chef/first-boot.json\" -E \"${var.env}\"" exited with non-zero exit status: 1
```

Thank you for taking a look!

**Comment by svanharmelen** [comment]
Thanks for the report! I also noticed this myself, and it looks like a generic issue with variables in provisioners as I can reproduce this with other provisioners as well.

Will have a look at this after the weekend...

**Comment by ivanst** [comment]
Awesome. Thank you.

**Comment by svanharmelen** [comment]
@ivanst it turns out we tested this incorrectly and that it was indeed a Chef provisioner specific bug which is now fixed in PR #2168

**Comment by ivanst** [comment]
Great! Thanks.
